unstract/core/tests/file_storage/test_fs_permanent.py

```diff
@@ -0,0 +1,111 @@
+import json
+import os.path
+
+import pytest
+from dotenv import load_dotenv
+from unstract.sdk.file_storage import FileStorageProvider
+from unstract.sdk.file_storage.fs_permanent import PermanentFileStorage
+
+load_dotenv()
+
+
+class TEST_CONSTANTS:
+    READ_FOLDER_PATH = os.environ.get("READ_FOLDER_PATH")
+    WRITE_FOLDER_PATH = os.environ.get("WRITE_FOLDER_PATH")
+    RECURSION_FOLDER_PATH = os.environ.get("RECURSION_FOLDER_PATH")
+    READ_PDF_FILE = os.environ.get("READ_PDF_FILE")
+    READ_TEXT_FILE = os.environ.get("READ_TEXT_FILE")
+    WRITE_PDF_FILE = os.environ.get("WRITE_PDF_FILE")
+    WRITE_TEXT_FILE = os.environ.get("WRITE_TEXT_FILE")
+    TEST_FOLDER = os.environ.get("TEST_FOLDER")
+    GCS_BUCKET = os.environ.get("GCS_BUCKET")
+    TEXT_CONTENT = os.environ.get("TEXT_CONTENT")
+    FILE_STORAGE_ENV = "FILE_STORAGE"
+
+
+def permanent_file_storage(provider: FileStorageProvider):
```

Comment:
Consider converting this into a `pytest.fixture` that can be used in subsequent tests rather than passing it in with `parameterize`. We have options to define this once only per test / module / session